segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -346,10 +352,9 @@ private void handleException(UUID writerId, long requestId, String segment, long
             log.warn("Bad attribute update by {} on segment {}.", writerId, segment, u);
             connection.send(new InvalidEventNumber(writerId, requestId, clientReplyStackTrace));
             connection.close();
-        } else if (u instanceof AuthenticationException) {
+        } else if (u instanceof TokenException) {
             log.warn("Token check failed while being written by {} on segment {}.", writerId, segment, u);
             connection.send(new WireCommands.AuthTokenCheckFailed(requestId, clientReplyStackTrace));
-            connection.close();
```

Comment:
Why are we removing the call to close the connection?

Reply:
Earlier, the connection was closed here, due to which the control was never reaching the `io.pravega.client.segment.impl.ResponseProcessor.authTokenCheckFailed(...)` method on the client side upon token-related failures on the segment store, which was a bug. The control was reaching the `ResponseProcessor.connectionDropped(...)` instead. As a result, upon failures in token verification, the client would hang due to subsequent retries and the corresponding exception would not be propagated to the client. 

I encountered this problem when running a new test `io.pravega.test.integration.DelegationTokenTest.testWriteFailsWhenTokenExpires(...)`.